src/esrs/postgres/mod.rs

```diff
@@ -207,23 +207,23 @@ impl<
             }
         }
 
-        let store_events: Vec<_> = events
-            .into_iter()
-            .map(|((event_id, event), sequence_number)| StoreEvent {
+        let mut store_events: Vec<StoreEvent<Evt>> = vec![];
+
+        for ((event_id, event), sequence_number) in events {
+            let store_event: StoreEvent<Evt> = StoreEvent {
                 id: event_id,
                 aggregate_id,
                 payload: event,
                 occurred_on,
                 sequence_number,
-            })
-            .collect();
-
-        for store_event in store_events.iter() {
-            let result = self.project_event(store_event, &mut connection).await;
-
-            if let Err(err) = result {
-                self.rollback(connection).await?;
-                return Err(err);
+            };
+
+            match self.project_event(&store_event, &mut connection).await {
+                Ok(()) => store_events.push(store_event),
+                Err(err) => {
+                    self.rollback(connection).await?;
```

Comment:
I would say so but this is a connection, not a transaction.. It was made in order to test stuff in projects that use this. It just open a transaction that never close, so that at the end of the tests the transaction is rollbacked and test database is still clean. Maybe there's a better way of do that?

Reply:
The [Pool](https://docs.rs/sqlx/latest/sqlx/pool/struct.Pool.html#method.begin) and the [Connection](https://docs.rs/sqlx/latest/sqlx/trait.Connection.html#tymethod.begin) already provide a `begin` method that does this and does a rollback unless you run `.commit()`.

Reply:
Yes, but that return a Transaction and if the transaction is rollbacked at the drop in the test you are running you'll not find any data in projectors/store. Like this is not possible to test anything

Reply:
Tbh I think it's better to avoid using transactions for testing.
sqlx 0.6.x already provides a nice way to do this https://docs.rs/sqlx/latest/sqlx/attr.test.html by creating an isolated db for each test (or how with separate schemas as we do it in platform-global/es-be https://github.com/primait/es-be/blob/master/src/test_helpers/src/test_pg_pool.rs but this works only for Postgres)

Reply:
Nice. Ok, next PR i'll do it